bassa.py

```diff
@@ -0,0 +1,535 @@
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def get_blocked_users_request(self, retries=1):
+        """Get all blocked user requests 
+        Args:
+            retires (int): number of retries upon failure
+        Returns:
+            response as json
+        """
+        endpoint = "/api/user/blocked"
+        api_url_complete = self.api_url + endpoint
+        while retries != -1:
+            result = requests.get(api_url_complete, headers=self.headers)
+            retries = retries-1
+            if int(result.status_code/100) in [3, 4]:
+                retries = 0
+                raise ResponseError(
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+                return result.json()
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def block_user_request(self, user_name=None, retries=1):
+        """Block a user request
+        Args:
+            user_name (str): Name of the user
+            retires (int): number of retries upon failure
+        Returns:
+            None
+        """
+        endpoint = "/api/user/blocked"
+        if user_name is None:
+            raise IncompleteParams
+        api_url_complete = self.api_url + endpoint + "/" + user_name
+        while retries != -1:
+            result = requests.post(api_url_complete, headers=self.headers)
+            retries = retries-1
+            if int(result.status_code/100) in [3, 4]:
+                retries = 0
+                raise ResponseError(
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def unblock_user_request(self, user_name=None, retries=1):
+        """Unblock a user request
+        Args:
+            user_name (str): Name of the user
+            retires (int): number of retries upon failure
+        Returns:
+            None
+        """
+        endpoint = "/api/user/blocked"
+        if user_name is None:
+            raise IncompleteParams
+        api_url_complete = self.api_url + endpoint + "/" + user_name
+        while retries != -1:
+            result = requests.delete(api_url_complete, headers=self.headers)
+            retries = retries-1
+            if int(result.status_code/100) in [3, 4]:
+                retries = 0
+                raise ResponseError(
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def get_downloads_user_request(self, limit=1, retries=1):
+        """Get downloads user request
+        Args:
+            limit (int): Number of records to return. limit 1 = 25 records
```

Comment:
What does limit 1 = 25 mean?

Reply:
It's according to our Bassa server deaign
So if limit is 1 then we can get upto 25 SQL records. 
Max Number of sql records = limit * 25